functions/lib/checkUserLessons.js

```diff
@@ -6,26 +6,25 @@ async function userCompletedCourse(userId, courseId) {
     .where('user_id', '==', userId)
     .get()
     .then(async (snapshot) => {
-      const docs = snapshot.docs.map((doc) => {
-        return doc.data().lesson
-      })
+      const docs = snapshot.docs.map((doc) => doc.data().lesson)
       const courseLessons = await db
         .collection('courses')
         .doc(courseId)
         .get()
         .then(async (courses) => {
           const course = courses.data()
-          return Object.keys(course.sections)
-            .map((section) => {
-              return course.sections[section].map((lesson) => {
-                return lesson.file
-              })
-            })
-            .flat()
+          // Find the first language that has sections
```

Comment:
this will break for existing courses unless you add `metadata` field and sections in all of them

Reply:
All current courses have been updated and now have the appropriate metadata fields